quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java

```diff
@@ -0,0 +1,187 @@
+            for (int width = 0; width < this.board[0].length; width++) {
+                this.board[height][width] = Cell.EMPTY;
+            }
+        }
+    }
+
+    private boolean checkPaths(Wall wall, Position[] positions, int[] heightsWin) {
+        Cell[][] newBoard = new Cell[this.board.length][this.board[0].length];
+        for (int height = 0; height < this.board.length; height++) {
+            System.arraycopy(this.board[height], 0, newBoard[height], 0, this.board[0].length);
+        }
+
+        newBoard = this.addWallToBoard(wall, newBoard);
+
+        for (int i = 0; i < positions.length; i++) {
+            if (this.checkingExistingPath(newBoard, positions[i], heightsWin[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //BFS
+    // 1 - checked
+    // 0 - not checked
+    private boolean checkingExistingPath(Cell[][] board, Position position, int rawExit) {
+        LinkedList<Position> nextToVisit
+                = new LinkedList<>();
+        nextToVisit.add(position);
+
+        while (!nextToVisit.isEmpty()) {
+            Position cur = nextToVisit.remove();
+
+            if (!this.isValidLocation(cur)
+                    || board[cur.getY()][cur.getX()] == Cell.CHECKED) {
+                continue;
+            }
+
+            // Exit - need to change for four players
+            if (cur.getY() == rawExit) {
+                return true;
+            }
+
+            //RIGHT
+            if (cur.getY() != this.board.length + 1 && board[cur.getY()][cur.getX() + 1] != Cell.WALL) {
+                Position Position
+                        = new Position(
+                        cur.getX() + 2,
+                        cur.getY());
+                nextToVisit.add(Position);
+            }
+            // LEFT
+            if (cur.getX() != 0 && board[cur.getY()][cur.getX() - 1] != Cell.WALL) {
+                Position Position
+                        = new Position(
+                        cur.getX() - 2,
+                        cur.getY());
+                nextToVisit.add(Position);
+            }
+            // UP
+            if (cur.getY() != this.board.length - 1 && board[cur.getY() + 1][cur.getX()] != Cell.WALL) {
+                Position Position
+                        = new Position(
+                        cur.getX(),
+                        cur.getY() + 2);
+                nextToVisit.add(Position);
+            }
+            // DOWN
+            if (cur.getY() != 0 && board[cur.getY() - 1][cur.getX()] != Cell.WALL) {
+                Position Position
+                        = new Position(
+                        cur.getX(),
+                        cur.getY() - 2);
+                nextToVisit.add(Position);
+            }
+            board[cur.getY()][cur.getX()] = Cell.CHECKED;
+        }
+        return true;
+    }
+
+    public Cell[][] addWallToBoard(Wall wall, Cell[][] board) {
```

Comment:
why the it gets and return the board??? (the board is a private member with getters and setters...)